sherpa-onnx/csrc/provider-config.h

```diff
@@ -0,0 +1,92 @@
+// sherpa-onnx/csrc/provider-config.h
+//
+// Copyright (c)  2024  Uniphore (Author: Manickavela)
+
+#ifndef SHERPA_ONNX_CSRC_ONNXRT_EXECUTION_PROVIDER_CONFIG_H_
+#define SHERPA_ONNX_CSRC_ONNXRT_EXECUTION_PROVIDER_CONFIG_H_
+
+#include <string>
+
+#include "sherpa-onnx/csrc/parse-options.h"
+#include "sherpa-onnx/csrc/macros.h"
+#include "onnxruntime_cxx_api.h"  // NOLINT
+
+namespace sherpa_onnx {
+
+struct CudaConfig {
+  uint32_t cudnn_conv_algo_search = OrtCudnnConvAlgoSearchHeuristic;
+
+  CudaConfig() = default;
+  CudaConfig(const uint32_t &cudnn_conv_algo_search)
```

Comment:
Please remove `const` and `&` for primitive types.

Reply:
sure, replicate this for Tensorrtconfig objects too for all the primitives?

Reply:
arguments of primitive type can be passed directly without `&`. Also we don't use `const` for primitive argments.